### ISSUE TITLE
Decompiler

### ISSUE THREAD

**Issue by Kelioser** [root-cause/v-decompiled-scripts#13]
which decompiler do you use? and what settings?

**Comment by root-cause** [comment]
I use a private fork of https://github.com/calamity-inc/GTA-V-Script-Decompiler, Here are my settings:
![image](https://github.com/root-cause/v-decompiled-scripts/assets/32650536/2cd099eb-275f-4c17-b4c2-a591fcf68db4)

**Comment by ghost** [comment]
> I use a private fork of https://github.com/calamity-inc/GTA-V-Script-Decompiler, Here are my settings: ![image](https://private-user-images.githubusercontent.com/32650536/343324560-2cd099eb-275f-4c17-b4c2-a591fcf68db4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0NDM2MTIsIm5iZiI6MTcxOTQ0MzMxMiwicGF0aCI6Ii8zMjY1MDUzNi8zNDMzMjQ1NjAtMmNkMDk5ZWItMjc1Zi00YzE3LWI0YzItYTU5MWZjZjY4ZGI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI2VDIzMDgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhODg3MWY3OTljMGVhNTUzNWE3YzMyNDJmZmQ2Yjg5MzkzNzFjMzkwZDAxN2MzZGNmYTIxZTFlOWRhZTUwMDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pOGV4WesmwE3YVU1Mh9QRAtj1AwhIZmryFQj5s5MPRQ)

why i dont see Show tunable Names 
![image](https://github.com/root-cause/v-decompiled-scripts/assets/138903739/98db36b9-9459-4a59-be44-53a3ba1d0e93)

**Comment by Kelioser** [comment]
yeah me too, i guess it's part of the reason his fork is private, oh well, it's a really good option to have especially how powerful it can be!

**Comment by ghost** [comment]
> yeah me too, i guess it's part of the reason his fork is private, oh well, it's a really good option to have especially how powerful it can be!

you are right

**Comment by iSk1P** [comment]
> yeah me too, i guess it's part of the reason his fork is private, oh well, it's a really good option to have especially how powerful it can be!

it would be great if there is an updated tutorial on how to do it because this one seems outdated https://www.unknowncheats.me/forum/grand-theft-auto-v/566606-decrypt-gta-scripts.html

**Comment by gir489returns** [comment]
> yeah me too, i guess it's part of the reason his fork is private, oh well, it's a really good option to have especially how powerful it can be!

In his credits he lists this repo from where he gets the tunable names from: https://github.com/Wildbrick142/V-Tunable-Names

The tunables are just a JOAAT like anything else, like the natives, the script event IDs, etc. The idea was that to squeeze every single ounce of performance out of the 360's integer-heavy CPU, they could instead compare 32-bit JOAATs than something like identifier. However, Rockstar sucks at coding, and thinks that writing a VM language is the best way to do scripts, when they could've just done something like implement Node.JS or Lua (like every other game engine), or Squirrel like Source 2. They instead had to be idiots and write it themselves.

All you need to do is basically override and implement the tunable code to do a JOAAT brute-force on the list. Although IDK if C# has an implementation of unordered_map (HashMap?), you could match the JOAAT hash to a string via the map. It should be very easy to implement.
